Source/SwiftLintFramework/Rules/MultilineParametersRule.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  MultilineParametersRule.swift
+//  SwiftLint
+//
+//  Created by Ornithologist Coder on 22/05/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct MultilineParametersRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "multiline_parameters",
+        name: "Multiline Parameters",
+        description: "Functions and methods parameters should be either on the same line, or one per line.",
+        nonTriggeringExamples: MultilineParametersRuleExamples.nonTriggeringExamples,
+        triggeringExamples: MultilineParametersRuleExamples.triggeringExamples
+    )
+
+    public func validate(file: File,
+                         kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard isValidFunction(kind),
+            let offset = dictionary.nameOffset,
+            let length = dictionary.nameLength
+            else {
+                return []
+        }
+
+        var numberOfParameters: Int = 0
+        var linesWithParameters: Set<Int> = []
+
+        for structure in dictionary.substructure {
+            guard
+                let structureOffset = structure.offset,
+                let structureKind = structure.kind, SwiftDeclarationKind(rawValue: structureKind) == .varParameter,
+                let (line, _) = file.contents.bridge().lineAndCharacter(forByteOffset: structureOffset)
+                else {
+                    continue
+            }
+
+            guard offset..<(offset + length) ~= structureOffset else { break }
```

Comment:
maybe use `continue` (and move to the `guard` above) so we don't rely on the order of parameters from SourceKit?